src/utils/xhr.js

```diff
@@ -1,369 +0,0 @@
-/**
- * Copyright 2018 The Subscribe with Google Authors. All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS-IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-import {assert} from './log';
-import {parseJson} from './json';
-import {parseUrl} from './url';
-import {utf8EncodeSync} from './bytes';
-
-/**
- * The "init" argument of the Fetch API. Currently, only "credentials: include"
- * is implemented.
- *
- * See https://developer.mozilla.org/en-US/docs/Web/API/GlobalFetch/fetch
- *
- * @typedef {{
- *   body: (!FormData|string|undefined),
- *   credentials: (string|undefined),
- *   headers: (!Object|undefined),
- *   method: (string|undefined),
- *   responseType: (string)
- * }}
- */
-export let FetchInitDef;
-
-/** @private @const {!Array<string>} */
-const allowedMethods_ = ['GET', 'POST'];
-
-/** @private @enum {number} Allowed fetch responses. */
-const allowedFetchTypes_ = {
-  document: 1,
-  text: 2,
-};
-
-/**
- * A class that polyfills Fetch API.
- */
-export class Xhr {
-  /**
-   * @param {!Window} win
-   */
-  constructor(win) {
-    /** @const {!Window} */
-    this.win = win;
-  }
-
-  /**
-   * We want to call `fetch_` unbound from any context since it could
-   * be either the native fetch or our polyfill.
-   *
-   * @param {string} input
-   * @param {!FetchInitDef} init
-   * @return {!Promise<!FetchResponse>|!Promise<!Response>}
-   * @private
-   */
-  fetch_(input, init) {
-    // TODO(avimehta): Should the requests go through when page is not visible?
-    assert(typeof input == 'string', 'Only URL supported: %s', input);
-    // In particular, Firefox does not tolerate `null` values for
-    // `credentials`.
-    const creds = init.credentials;
-    assert(
-      creds === undefined || creds == 'include' || creds == 'omit',
-      'Only credentials=include|omit support: %s',
-      creds
-    );
-    // Fallback to xhr polyfill since `fetch` api does not support
-    // responseType = 'document'. We do this so we don't have to do any parsing
-    // and document construction on the UI thread which would be expensive.
-    if (init.responseType == 'document') {
```

Comment:
Do we need to preserve this behavior? responseType in init is not part of the fetch standard AFAICT.

Reply:
Swgjs only uses `fetch` in a few places, through the Fetcher. In those places, I didn't see any evidence we needed this behavior